roles/slurm/tasks/controller.yml

```diff
@@ -1,20 +1,45 @@
 ---
 
-- name: install mariadb
+- name: install dependencies
   apt:
     name: "{{ item }}"
     state: latest
   with_items:
     - mariadb-server
     - python-mysqldb
+    - s-nail
+    - ssmtp
+  when: ansible_os_family == "Debian"
 
-- name: install s-nail and ssmtp
-  apt:
+- name: install dependencies
+  yum:
     name: "{{ item }}"
     state: latest
   with_items:
-    - s-nail
+    - mariadb-server
+    - MySQL-python
+    - mailx
     - ssmtp
+  when: ansible_os_family == "RedHat"
+
+- name: edit mariadb selinux policy
+  shell: grep mysqld /var/log/audit/audit.log | audit2allow -M mariadb_local
+  args:
+    creates: mariadb_local.pp
+  register: policy_created
+  when: ansible_os_family == "RedHat"
+
+- name: apply mariadb selinux policy
```

Comment:
This task fails for me because it can't find the mariadb_local.pp file.

```
TASK [slurm : apply mariadb selinux policy] ************************************************************************
fatal: [virtual-login]: FAILED! => {"changed": true, "cmd": ["semodule", "-i", "mariadb_local.pp"], "delta": "0:00:00.337222", "end": "2019-03-21 21:10:44.469524", "msg": "non-zero return code", "rc": 1, "start": "2019-03-21 21:10:44.132302", "stderr": "libsemanage.map_file: Unable to open mariadb_local.pp\n (No such file or directory).\nlibsemanage.semanage_direct_install_file: Unable to read file mariadb_local.pp\n (No such file or directory).\nsemodule:  Failed on mariadb_local.pp!", "stderr_lines": ["libsemanage.map_file: Unable to open mariadb_local.pp", " (No such file or directory).", "libsemanage.semanage_direct_install_file: Unable to read file mariadb_local.pp", " (No such file or directory).", "semodule:  Failed on mariadb_local.pp!"], "stdout": "", "stdout_lines": []}
```

When I manually run the previous command on the host where this failed, it shows "Nothing to do". Checking the `audit.log` file shows no mentions of mysqld.

```
[root@virtual-login ~]# grep mysqld /var/log/audit/audit.log | audit2allow -M mariadb_local
Nothing to do
```

Suggest that you check for existence of the `mariadb_local.pp` file before attempting to run `semodule`. Maybe something like this?

```
shell: ls mariadb_local.pp && semodule -i mariadb_local.pp
```

Reply:
@ajdecon fixed this one using a proper SELinux "thingy", please try again